sdk/identity/identity/src/credentials/azureCliCredential.ts

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { TokenCredential, GetTokenOptions, AccessToken } from "@azure/core-http";
+import { createSpan } from "../util/tracing";
+import { AuthenticationErrorName } from "../client/errors";
+import { CanonicalCode } from "@opentelemetry/types";
+import { logger } from "../util/logging";
+
+import * as child_process from "child_process";
+
+export interface CredentialClient {
+  getAzureCliAccessToken(resource: string): Promise<any>;
+}
+
+class AzureCliCredentialClient implements CredentialClient {
+  public getAzureCliAccessToken(resource: string) {
+    return new Promise((resolve, reject) => {
+      try {
+        child_process.exec(
+          `az account get-access-token --output json --resource ${resource}`,
+          (error, stdout, stderr) => {
```

Comment:
I think this interface and class should return an `AccessToken`, not the pipes from running the command.  It'd make mocking and usage easier if it was just handled in the "client" class.  What do you think?

Reply:
I don't mind the stdout/stderr approach since it's mocking around the call out to the `az`.